pkg/stats/collector.go

```diff
@@ -104,30 +103,29 @@ func (t *TimeTicker) Tick() <-chan time.Time {
 
 type BufferedCollector struct {
 	cache            keyIndex
-	writes           chan Metric
+	updatesCh        chan Metric
 	sender           Sender
-	sendTimeout      time.Duration
 	flushTicker      FlushTicker
 	flushSize        int
 	heartbeatTicker  FlushTicker
 	installationID   string
 	processID        string
 	runtimeCollector func() map[string]string
 	runtimeStats     map[string]string
-	pendingWrites    sync.WaitGroup
-	pendingRequests  sync.WaitGroup
-	ctxCancelled     int32
-	done             chan bool
-	runCalled        int32
+	inFlight         sync.WaitGroup
+	closed           int32
 	extended         bool
 	log              logging.Logger
+	wg               sync.WaitGroup
+	cancel           context.CancelFunc
+	sendCh           chan *InputEvent
```

Comment:
Seems like we added or replaced many synchronization protocols. Can you document each one for the next developer?

Reply:
added comments